components/econet/econet.h

```diff
@@ -74,7 +76,20 @@ class Econet : public Component, public uart::UARTDevice {
 
   void set_src_address(uint32_t address) { src_adr_ = address; }
   void set_dst_address(uint32_t address) { dst_adr_ = address; }
-  void set_update_interval(uint32_t interval_millis) { update_interval_millis_ = interval_millis; }
+  void set_mod_req_updates(std::vector<uint8_t> keys, std::vector<uint32_t> values) {
+    for (auto i = 0; i < keys.size(); i++) {
+      request_mod_update_interval_millis_[keys[i]] = values[i];
+      min_update_interval_millis_ = std::min(min_update_interval_millis_, values[i]);
+    }
+  }
+  void set_update_interval(uint32_t interval_millis) {
+    update_interval_millis_ = interval_millis;
+    min_update_interval_millis_ = interval_millis;
+    for (auto i = 0; i < this->request_mod_update_interval_millis_.size(); i++) {
+      request_mod_update_interval_millis_[i] = update_interval_millis_;
+      request_mod_last_requested_[i] = i * update_interval_millis_ / MAX_REQUEST_MODS;
```

Comment:
Similarly this won't be initialized if set_update_interval is never called. I think I left a comment earlier. I don't see any value initializing this vector. I'd just leave it at 0. I don't see the point initializing to these strange values. You are using at line 437:
`if ((this->loop_now_ - request_mod_last_requested_[request_mod]) > request_mod_update_interval_millis_[request_mod])`

Since now is so much larger this will be true at the beginning, similarly to if you had them at 0.
I guess you meant to have `millis() - i * update_interval_millis_ / MAX_REQUEST_MODS` or something similar but I really don't see the point complicating the code.

Reply:
Removed. Forgot to remove this from the last review.